### PR TITLE
NormalizerNFKC*: Add tests for mixing katakana options

### PR DESCRIPTION
**Pull request by HashidaTKS**
Added tests for mixing katakana options.
And fixed `unify_katakana_trailing_o` in order to work fine for the following cases.

1. `ョー -> ヨー -> ヨオ -> ヨウ`
  
   `ョー` was normalized to `ヨオ` (should be `ヨウ`) because `unify_katakana_trailing_o` was applied before `unify_kana_case` and `unify_katakana_trailing_o` is not normalized `オ` after small letters.
  I have added `オ` after `ォ` or `ョ` as a target of `unify_katakana_trailing_o`.

2. `ヺー -> ヺオ -> ヺウ`

   `ヺー` was normalized to `ヺオ` because  `unify_katakana_trailing_o` didn't target `ヺ`.
   I have added `ヺ` as a target of `unify_katakana_trailing_o`.

### CODE REVIEW

File: test/command/suite/normalizers/nfkc100/mix/katakana/prolonged_sound_mark/du_small.expected [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      "katakana",
+      "katakana",
+      "katakana",
+      "katakana"
+    ],
+    "checks": [
+      6,
+      0,
+      0,
+      32616,
```
Comment (HashidaTKS) [inline test/command/suite/normalizers/nfkc100/mix/katakana/prolonged_sound_mark/du_small.expected:26]: This is odd, this should be 3 but 32616.
The value 32626 is not stable, changes each time I run it.
I will check it.

Comment (kou) [reply test/command/suite/normalizers/nfkc100/mix/katakana/prolonged_sound_mark/du_small.expected:26]: あぁ、昨日私がいじったところが悪いかも。

Comment (HashidaTKS) [reply test/command/suite/normalizers/nfkc100/mix/katakana/prolonged_sound_mark/du_small.expected]: 「ズァ」と「ヅァ」のときだけ発生していて、「ヴァ」などでは起きないので、昨日の須藤さんの修正とは関係なさそうな感じもします。（調べています。）

Comment (HashidaTKS) [reply test/command/suite/normalizers/nfkc100/mix/katakana/prolonged_sound_mark/du_small.expected]: 最新のmasterにrebaseしたところ直ったので、関係あったようです。

Comment (kou) [reply test/command/suite/normalizers/nfkc100/mix/katakana/prolonged_sound_mark/du_small.expected]: さーせん。。。

Comment (HashidaTKS) [reply test/command/suite/normalizers/nfkc100/mix/katakana/prolonged_sound_mark/du_small.expected]: いえ、ご対応ありがとうございます。

**Comment by HashidaTKS** [comment]
@kou 

Would you review this when you have time?

File: lib/normalizer.c [hunk 2]
```diff
@@ -2797,30 +2799,30 @@ grn_nfkc_normalize_unify(grn_ctx *ctx,
...
       grn_nfkc_normalize_context_rewind(ctx, &unify);
     }
-    bool need_trailing_check = false;
-    grn_nfkc_normalize_unify_stateful(ctx,
-                                      data,
-                                      &unify,
-                                      grn_nfkc_normalize_unify_katakana_trailing_o,
-                                      &need_trailing_check,
-                                      "[unify][katakana-trailing-o]");
+    grn_nfkc_normalize_unify_stateless(ctx, data, &unify, false);
```
Comment (kou) [inline lib/normalizer.c:2807]: I don't want to change the current following order:

1. `grn_nfkc_normalize_unify_stateless(before)`
2. `grn_nfkc_normalizer_unify_statefull()`s
3. `grn_nfkc_normalize_unify_stateless(after)`

For keeping unifying order simple.

How about adding `ョ` and small letters to `katakana_trailing_o` targets.

**Review by kou — COMMENTED** [review]
Hmm. Adding similar tests to all `test/command/suite/normalizers/nfkc*/` is annoying... We may be able to unify them by improving grntest in the future...

File: lib/normalizer.c (hunk 2 above)
Comment (HashidaTKS) [reply lib/normalizer.c:2807]: Ok, I will fix so as to normalize `オ` after small letters.

**Comment by HashidaTKS** [comment]
> Hmm. Adding similar tests to all `test/command/suite/normalizers/nfkc*/` is annoying... We may be able to unify them by improving grntest in the future...

Agree.

File: lib/normalizer.c (hunk 2 above)
Comment (HashidaTKS) [reply lib/normalizer.c]: Fixed.

**Comment by HashidaTKS** [comment]
@kou 

Thank you, I have addressed your comments.

File: test/command/suite/normalizers/nfkc100/mix/katakana/di.test [hunk 3]
```diff
@@ -0,0 +1,14 @@
+normalize \
+  'NormalizerNFKC100("unify_katakana_v_sounds", true, \
+                    "unify_katakana_gu_small_sounds", true, \
```
Comment (kou) [inline test/command/suite/normalizers/nfkc100/mix/katakana/di.test:3]: Broken indent but I'll merge this as-is. I can't review all tests...

**Review by kou — COMMENTED** [review]
I can't review all tests but I merge this.

**Pull request merged**